set.mm

```diff
@@ -496507,6 +496553,132 @@ The sumset operation can be used for both group (additive) operations and
+        ZUEZUFYSBUHRZFUIOUUAYQSUUBYRUUAYQCUJUKAYQDUHRZYPULZUUCAHUUDTZYQUUETPUUD
+        HYPDGUCUMUNUOAYPDBUUDUIUPBDYPUQQSANURAUUDUSADGUCVEACDEUTQZTZDUPTZLDECVA
+        ZUOVBVCAYRUIACYQUUGLVDVFVGAYSYTVHZUFYTAYRUUKAUAYRUUKAUAUGZYRTUBUGZCRZUU
+        LSZUBYQVIZUULUUKTZAUBUUDYQUULCAUUDEUHRZCAUUHUUDUURCVRZLDECUUDUURUUDVJZU
+        URVJZVKZUOVLZAHYPUUDAGDVMRZTZGDVNRTZUUDYPVOZAGCVPIVHUEZUVDMAUUHUVHUVDTL
+        DECIKVQUOVSZGDVTZYPDUUDGUUTYPVJZWAWBZWIZWCAUUPUULYTSZUUQAUUPUVNAUUOUVNU
+        BYQAUUMYQTZWDZUUOWDUUNUULYTUVPUUOWEUVPUUNYTSUUOUVPYTHDWFRZRZCRZUUNEWGRZ
+        QZUVTQZYTIUVTQZUUNYTUVPUWAIYTUVTUVPUVRUUMDWGRZQZCRZUWAIUVPUUHUVRUUDTUUM
+        UUDTZUWFUWASAUUHUVOLWHZUVPUUDDUVQHUUTUVQVJZUVPUUHUUIUWHUUJUOZAUUFUVOPWH
+        ZWJAYQUUDUUMUVMWKZUWDUVTDEUVRCUUMUUDUUTUWDVJZUVTVJZWLWMUVPUWEUUDTZUWFIS
+        ZUVPCUUDWNZUWEUVHTZUWOUWPWDZAUWQUVOUVCWHUVPUWEGUVHUVPUUIGUUDWOZHUUMYPWP
+        ZUWEGTZUWJAUWTUVOAUVEUVFUWTUVIUVJUUDGDUUTWQWBWHAUUFUVOUXAPUUFUVOUXAUUMU
+        ITUUFUVOUXAWRUBXAUUMHYPUIUUDWSXBWTXKUUIUWTWDZUXAWDUUFUWGUXBUXCUXAUUFUWG
+        UXBYJHUUMUWDYPGDUVQUPUUDUUTUWIUWMUVKXCWTXDXEMXFUWQUWRUWSUUDIUWECXGWTXHX
+        IXJXLUVPUWBYTYTEWFRZRZUUNUVTQZUVTQZUUNUVPUWAUXFYTUVTUVPUVSUXEUUNUVTUVPU
+        UHUUFUVSUXESUWHUWKUUDDECUVQUXDHUUTUWIUXDVJZXMXHXNXLUVPEUPTZYTUURTZUUNUU
+        RTUXGUUNSUVPUUHUXIUWHDECXOUOZUVPUUDUURHCUVPUUHUUSUWHUVBUOZUWKXPZUVPUUDU
+        URUUMCUXLUWLXPUURUVTEUXDYTUUNUVAUWNUXHXQWMXTUVPUXIUXJUWCYTSUXKUXMUURUVT
+        EYTIUVAUWNKXRXHYAWHXJXSAUVNWDZUUOYTUULSZUBHYQAHYQTZUVNAUVGUUFUXPUVLPHYP
+        UUDYBXHWHUUMHSUUOUXOWRUXNUUMHUULCYCYDUXNUULYTAUVNWEYKYEYFUAYTYGYHYIYLUK
+        YTHCYMYNYOXT $.
+    $}
+
+    ${
+      ghmquskerlem2.y $e |- ( ph -> Y e. ( Base ` Q ) ) $.
+      $( Lemma for ~ ghmqusker (Contributed by Thierry Arnoux, 14-Feb-2025.) $)
+      ghmquskerlem2 $p |- ( ph -> E. x e. Y ( J ` Y ) = ( F ` x ) ) $=
+        ( co cfv wcel cv cqg cec wceq cbs wrex cqs cvv cgrp cqus a1i eqidd cghm
+        ovexd ghmgrp1 syl qusbas eleqtrrd elqsg biimpa syl2anc wa wer csubg csn
+        ccnv cima cnsg ghmker nsgsubg 3syl eqeltrid eqger ad2antrr simplr ecref
+        eqid simpr fveq2d ghmquskerlem1 eqtrd jca expl reximdv2 mpd ) AIBUAZEHU
+        BRZUCZUDZBEUESZUFZIGSZWFDSZUDZBIUFAICUESZTZIWJWGUGZTZWKQAIWOWQQAWGECWJU
+        HUICEWGUJRUDAOUKAWJULAEHUBUNADEFUMRTZEUITMEFDUOUPUQURWPWRWKBWJIWGWOUSUT
+        VAAWIWNBWJIAWFWJTZWIWFITZWNVBAWTVBZWIVBZXAWNXCWFWHIXCWJWGVCZWTWFWHTAXDW
+        TWIAHEVDSZTXDAHDVFJVEVGZXENAWSXFEVHSTXFXETMEFDJLVIXFEVJVKVLWGEWJHWJVQWG
+        VQVMUPVNAWTWIVOZWFWGWJVPVAXBWIVRZURXCWLWHGSWMXCIWHGXHVSXCCDEFGHWFJKLAWS
+        WTWIMVNNOPXGVTWAWBWCWDWE $.
+    $}
+
+    ghmqusker.s $e |- ( ph -> ran F = ( Base ` H ) ) $.
+    $( The mapping ` J ` induced by a surjective group homomorphism ` F ` from
+       the quotient group ` Q ` over ` F ` 's kernel ` K ` is a group
+       isomorphism.  In this case, one says that ` F ` factors through ` Q ` ,
```

Comment:
To say that `F factors through Q` one should prove that the composition of the natural homomorphism G -> Q and J equals F. I think this is (almost?) proved in `ghmquskerlem1`, so maybe it's better to provide the missing statement or reference it in the comment.

Reply:
Yes, that proof is quite easy given `ghmquskerlem1 `. I've added it as `ghmquskerco`.
That fact does not help to prove we have an isomorphism, does it?

Reply:
I guess it does help. Since `F = ( J o. L )` and `F` is a surjection, you immediately get that `J` is a surjection. Let's say `J(x) = J(y)` for some `x` and `y`. Since `L` is a surjection we have `x = L(a)`, `y = L(b)` for some `a` and `b`, hence `J(L(a)) = J(L(b))` and `F(a) = F(b)`. Since `L` and `F` have the same kernel, we have `L(a) = L(b)`, i.e. `x = y` and `J` is injective.

To show that `J` is a homomorphism, again take `x = L(a)`, `y = L(b)` and notice that
`J(x)*J(y) = J(L(a))*J(L(b)) = F(a)*F(b) = F(a*b) = J(L(a*b)) = J(L(a)*L(b)) = J(x*y)`.

Having written all this I'll admit that I haven't checked the proof of `ghmqusker` so maybe this won't shorten the argument at all :-)

Reply:
To show that `J` is a homomorphism takes a big part of the proof, I think it would be roughly the same length.
For injectivity I use [`~ kerf1ghm`](https://us.metamath.org/mpeuni/kerf1ghm.html) which already existed.
So only the surjection part might get shorter (but you also need the fact that `L` is a surjection too) [`~ foco`](https://us.metamath.org/mpeuni/foco.html) could be used.